### PR TITLE
Clarification, tests, refactoring and standards

### PR DESCRIPTION
**Pull request by ghophp**
Hello guys, sorry for intrude into this code base like this, but I was willing to practice some Go and saw this attempt to create a layer to ease the REST process over `go-json-rest` so I just gave it a try :neckbeard: 

**Clarify** :books: 

As I had to setup the project myself, I created the README to help the future users and contributors to fast dive into code. Both the README from the `example-api` and the main project were updated. The example-api got more instructions to allow no mistake doing that simple steps, and the core README gave explanations, basic sections like installation and usage.

**Tests** :ok_hand: 

As I was willing to learn more about the code base, I started creating a test suit that pushed me into the code understanding before changing anything, so you can see now a `entity_api_test.go` file on the api package, and this file uses the `util.go` from test package. I got this standard from the `go-json-rest` library.

**Standards** :eyes:  

I was advancing on the tests suits writing the tests that match the standards, and found some different situations that pushed me to some refactoring. So let's explain step by step:

```
entity_api.go - GetEntity - 404 instead of 500
```

The `/api/:entity/:id` call instead of return a 404 if the entity was not found, was returning 500, which is not right according to the REST standards, to make easy for clients, the better thing to do is use the proper status code.
 
```
entity_api.go - PostEntity - 201 instead of 200
```

Again, the standards say that the better to do regarding the POST method, is to use 201 (Created) when you successfully create an entity. Also, is important to send a the header `Location` with the new URL to query the resource, both were implemented and test covered.

```
entity_api.go - PutEntity - Differ between 200, 204 and 404.
```

When updating an entity, the standards say that the best to do is: when a entity is updated (rows affected) the code is 200, if the entity was found but no rows affected 204 (no content). And finally if the entity doesn't exits, the API should return 404. The changes were made and the tests are covering the cases.

**Refactoring** :+1: 

```
entity_manager.go - No more extra loop
```

In the entity_manager I basically introduced a better way of creating the strings for the queries, instead of doing an extra loop with an auxiliar variable and a if just to put comma or not, we are using now an array of strings and a join from the strings pkg which is much more elegant and reliable, removing 3 loops from the code.

```
entity_manager.go and entity_api - constants
```

I have introduced a few constants aiming to remove the magic values running around, so we have a more clear and centered code to check for the standards. Also, in the future, if we plan to make this more dynamic, like allowing the user to chose the default limit, offset, id column and direction, this will be easier to refactor.

### CODE REVIEW

**Comment by dirkaholic** [comment]
@ghophp Wow, will have a look into it later.

**Comment by ghophp** [comment]
@dirkaholic did you took some time to review that?

File: api/entity_api.go [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 	"net/http"
-	"fmt"
 	"strconv"
-	eram "github.com/Onefootball/entity-rest-api/manager"
 )
 
+const ID_COLUMN string = "id" //TODO: this can be dynamic and should
+const OFFSET string = "0"
+const LIMIT string = "10"
+const ORDER_DIR string = "ASC"
```
Comment (dirkaholic) [inline api/entity_api.go]: I think all these things should be dynamic so a constant is probably not the best thing to use but can be changed when making them dynamic.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
```
Comment (dirkaholic) [inline README.md]: Docs are :+1:

File: api/entity_api_test.go [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+
+	if err != nil {
+		t.Error(err)
+	} else {
+
+		if len(data) < 2 {
+			t.Errorf("Should have found at least 2 tags. %v", data)
+		}
+	}
+}
```
Comment (dirkaholic) [inline api/entity_api_test.go]: Test is :+1: as well

File: manager/entity_manager.go [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
 	"fmt"
+	"strconv"
 	"strings"
 	"time"
-	"errors"
-	"strconv"
-	"database/sql"
 )
 
+const ID_COLUMN string = "id" //TODO: this can be dynamic and should
```
Comment (dirkaholic) [inline manager/entity_manager.go]: This is a duplicate and should go to a more central place from my point of view.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
+entity-rest-api
+===============
+
+Description
+-----------
+
+entity-rest-api aims to offer a abstract layer over the `go-json-rest` repository, allowing your project to easily query using REST standards to do it.
```
Comment (dirkaholic) [inline README.md]: I would also mention that it can be used directly together with ng-admin. I can add that in a later PR.

**Pull request merged**